model/Hierarchy.php

```diff
@@ -634,15 +655,26 @@ public function liveChildren($showAll = false, $onlyDeletedFromStage = false) {
 		}
 
 		$baseClass = ClassInfo::baseDataClass($this->owner->class);
-		$children = $baseClass::get()
-			->filter('ParentID', (int)$this->owner->ID)
-			->exclude('ID', (int)$this->owner->ID)
-			->setDataQueryParam(array(
-				'Versioned.mode' => $onlyDeletedFromStage ? 'stage_unique' : 'stage',
-				'Versioned.stage' => 'Live'
-			));
-
-		if(!$showAll) $children = $children->filter('ShowInMenus', 1);
+		$hidden = $this->owner->config()->hide_from_hierarchy;
+		if (empty($hidden)) {
+			$children = $baseClass::get()
+				->filter('ParentID', (int)$this->owner->ID)
+				->exclude('ID', (int)$this->owner->ID)
+				->setDataQueryParam(array(
+					'Versioned.mode' => $onlyDeletedFromStage ? 'stage_unique' : 'stage',
```

Comment:
Same as above; Don't duplicate all of the logic if only a small part of your code changes depending on `$hidden` :)